include/z64save.h

```diff
@@ -4,6 +4,25 @@
 #include "ultra64.h"
 #include "z64math.h"
 
+typedef enum {
+    /*  0 */ HUD_VISIBILITY_IDLE,
+    /*  1 */ HUD_VISIBILITY_NONE,
+    /*  2 */ HUD_VISIBILITY_NONE_ALT, // Identical to HUD_VISIBILITY_NONE
+    /*  3 */ HUD_VISIBILITY_HEARTS_WITH_OVERWRITE, // Uses Interface_UpdateButtonAlphas so gives the opportunity to dim only disabled buttons
+    /*  4 */ HUD_VISIBILITY_A,
+    /*  5 */ HUD_VISIBILITY_A_HEARTS_MAGIC_WITH_OVERWRITE, // Uses Interface_UpdateButtonAlphas so gives the opportunity to dim only disabled buttons
+    /*  6 */ HUD_VISIBILITY_A_HEARTS_MAGIC_MINIMAP_WITH_OVERWRITE, // Uses Interface_UpdateButtonAlphas so gives the opportunity to dim only disabled buttons
+    /*  7 */ HUD_VISIBILITY_ALL_NO_MINIMAP_W_DISABLED, // Only raises button alphas if not disabled
```

Comment:
Admittedly, the distinction between `HUD_VISIBILITY_ALL_NO_MINIMAP` and `HUD_VISIBILITY_ALL_NO_MINIMAP_W_DISABLED` comes from MM where both cases exist, and the only difference between then is that:

`HUD_VISIBILITY_ALL_NO_MINIMAP`
        -> increases alphas directly in `Interface_UpdateHudAlphas`

`HUD_VISIBILITY_ALL_NO_MINIMAP_W_DISABLED`
        -> Increases alphas indirectly through `Interface_RaiseButtonAlphas`

So the difference between then is increasing button alphas independent of button status, and increasing button alphas depended on button status. No other state besides `HUD_VISIBILITY_ALL` will consider button status unless `hudVisibilityForceButtonAlphasByStatus` is set

Reply:
I'd argue `HUD_VISIBILITY_ALL_NO_MINIMAP` should just be the name used in OoT since it doesn't have the distinction, but it's probably fine regardless since this mode only ever used once and internally

Reply:
I did change this to `HUD_VISIBILITY_ALL_NO_MINIMAP_WITH_BTN_STATUS` instead, but I'd also be fine with just `HUD_VISIBILITY_ALL_NO_MINIMAP` if we wanted to